Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm

```diff
@@ -289,9 +289,9 @@ static Boolean isAXAuthenticatedCallback(audit_token_t auditToken)
     SandboxExtension::consumePermanently(parameters.trustdExtensionHandle);
 #endif // PLATFORM(MAC)
 #if USE(APPLE_INTERNAL_SDK)
-    if (parameters.restrictImageAndVideoDecoders) {
-        ImageDecoderCG::enableRestrictedDecoding();
-        restrictImageAndVideoDecoders();
+    if (parameters.enableDecodingHEIC) {
+        ImageDecoderCG::enableDecodingHEIC();
+        enableDecodingHEIC();
```

Comment:
Note to all readers, including myself: enableDecodingHEIC() does three things:
(1) Adds HEIC to the list of allowed video formats. (HEIC is absent by default. Also, adding HEIC as a video format does not add it as an image format. A separate call to setAdditionalSupportedImageTypes is required for that.)
(2) Removes a bunch of other video formats from the list of allowed video formats.
(3) Asks Video Toolbox, if invoked, to run in-process without use of hardware devices or IOSurface.

Reply:
Given this behavior, I think a nice follow-up would be to rename enableDecodingHEIC() again (sorry!) like so:

setVideoDecodingBehaviors(bool allowHEIC, bool allowIOKit)